src/System.Linq.Queryable/src/System/Linq/Queryable.cs

```diff
@@ -842,6 +842,23 @@ public static TSource SingleOrDefault<TSource>(this IQueryable<TSource> source,
                     ));
         }
 
+        public static bool TrySingle<TSource>(this IQueryable<TSource> source, out TSource element)
+        {
+            Tuple<bool, TSource> result = TrySingle(source);
+            element = result.Item2;
+            return result.Item1;
+        }
+
+        internal static Tuple<bool, TSource> TrySingle<TSource>(this IQueryable<TSource> source)
+        {
+            if (source == null)
+                throw Error.ArgumentNull(nameof(source));
+            return source.Provider.Execute<Tuple<bool, TSource>>(
+                Expression.Call(
+                    null,
+                    CachedReflectionInfo.TrySingle_TSource_1(typeof(TSource)), source.Expression));
+        }
+
```

Comment:
TrySingle is different from other methods that it return boolean result and return out parameter. I can not create `Funct<T1, T2, TResult>` with out parameter. `Funct<T1, out T2, TResult>` is wrong for compiler.
Now I am waiting answer from @bartdesmet on this issue.